### DISCUSSION TITLE
Defining a function

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#23647] [Q&A Modules: General] [answered]
Hello everyone,

Is there any way to define a function in MOOSE based on solution values in quadrature points?

In my project, defining the function based on DerivativeParsedMaterial will lead to a high error because the value of function is constant inside each element.

Thanks,
Ali

**Comment by GiudGiud** [comment, marked as answer]
Hello

Why dont you want to just use a variable to represent that solution at "Qp points"?

I think the SolutionFunction might be the best shot here for that otherwise
https://mooseframework.inl.gov/source/functions/SolutionFunction.html

Guillaume
